File: leetcode/maximum_twin_sum_of_linked_list/maximum_twin_sum_of_linked_list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    int pairSum(ListNode* head) {
        vector<int> listNode;
        for(ListNode* i=head;i!=nullptr;i=i->next){
            int val=(i->val);
            listNode.push_back(val);
        }

        int n=listNode.size();
        int ans=-1;
        for(int i=0;i<n/2;i++){
            int twinSum=listNode[i]+listNode[n-1-i];
            if(twinSum>ans){
                ans=twinSum;
            }
        }
        return ans;
    }
};

File: leetcode/maximum_twin_sum_of_linked_list/maximum_twin_sum_of_linked_list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        iter_list = head
        pure_list=[]
        while iter_list!=None:
            val=iter_list.val
            pure_list.append(val)
            iter_list=iter_list.next
        
        n=len(pure_list)

        ans=0
        for i in range(n//2):
            left=pure_list[i]
            right=pure_list[n-i-1]

            twin_sum=left+right
            if ans<twin_sum:
                ans=twin_sum
        
        return ans
